FSQLocationBroker/FSQLocationBroker.m

```diff
@@ -516,13 +516,13 @@ - (void)locationManager:(CLLocationManager *)manager didUpdateLocations:(NSArray
     BOOL isBackgrounded = applicationIsBackgrounded();
     CLLocation *mostRecentLocation = [locations firstObject];
     for (CLLocation *location in locations) {
-        if ([mostRecentLocation.timestamp laterDate:location.timestamp] == location.timestamp) {
+        if ([mostRecentLocation.timestamp compare:location.timestamp] == NSOrderedAscending) {
             mostRecentLocation = location;
         }
     }
     
     // only update if currentLocation is not set or if mostRecentLocation happened after currentLocation
-    if (!self.currentLocation || [self.currentLocation.timestamp laterDate:mostRecentLocation.timestamp] == mostRecentLocation.timestamp) {
+    if (!self.currentLocation || [self.currentLocation.timestamp compare:mostRecentLocation.timestamp] == NSOrderedAscending) {
```

Comment:
Technically this could still have undefined behavior if `locations` is `nil` or `@[]` because then this could wind up comparing a defined `NSDate` to `nil`, but I agree the risk is very minimal.